test/solvers/functional/default_functionals_test.py

```diff
@@ -0,0 +1,319 @@
+    cc_cc_func = cc_func.convex_conj
+    assert isinstance(cc_cc_func, type(expected_functional))
+
+
+def test_L2_norm():
+    """Test the L2-norm."""
+
+    n = 10
+    space = odl.rn(n)
+    func = odl.solvers.L2Norm(space)
+    x = noise_element(space)
+
+    # Evaluation of the functional
+    expected_result = np.sqrt(np.sum(x**2))
+    assert almost_equal(func(x), expected_result, places=PLACES)
+
+    # The gradient
+    if x.norm() == 0:
+        # In this case the gradient is not defined
+        with pytest.raises(ValueError):
+            func.gradient
+    else:
+        expected_result = x / x.norm()
+        assert all_almost_equal(func.gradient(x), expected_result,
+                                places=PLACES)
+
+    # The proximal operator
+    sigma = np.random.rand()
+    # Explicit computation: x * (1 - sigma/||x||) if ||x|| > 1, 0 else
+    norm_less_than_sigma = 0.9 * sigma / n * space.one()
+    assert all_almost_equal(func.proximal(sigma)(norm_less_than_sigma),
+                            space.zero(), places=PLACES)
+
+    norm_larger_than_sigma = 1.1 * x.norm() * x
+    expected_result = (norm_larger_than_sigma *
+                       (1.0 - sigma / norm_larger_than_sigma.norm()))
+    assert all_almost_equal(func.proximal(sigma)(norm_larger_than_sigma),
+                            expected_result, places=PLACES)
+
+    # The convex conjugate functional
+    cc_func = func.convex_conj
+
+    # Evaluation of convex conjugate
+    # Explicit calculation: 0 if ||x|| < 1, infty else
+    norm_larger_than_one = 2.0 * space.one()
+    assert cc_func(norm_larger_than_one) == np.inf
+
+    norm_less_than_one = 0.9 * 1.0 / n * space.one()
+    assert cc_func(norm_less_than_one) == 0
+
+    # The gradient of the convex conjugate (not implemeted)
+    with pytest.raises(NotImplementedError):
+        cc_func.gradient
+
+    # The proximal of the convex conjugate
+    # Explicit computation: x/||x||
+    expected_result = x / x.norm()
+    assert all_almost_equal(cc_func.proximal(sigma)(x), expected_result,
+                            places=PLACES)
+
+    # The biconjugate, which is the functional itself since it is proper,
+    # convex and lower-semicontinuous
+    expected_functional = odl.solvers.L2Norm(space)
+    cc_cc_func = cc_func.convex_conj
+    assert isinstance(cc_cc_func, type(expected_functional))
+
+
+def test_L2_norm_squared():
+    """Test the squared L2-norm."""
+
+    n = 10
+    space = odl.rn(n)
+    func = odl.solvers.L2NormSquared(space)
+    x = noise_element(space)
+
+    # Evaluation of the functional
+    expected_result = np.sum(x**2)
+    assert almost_equal(func(x), expected_result, places=PLACES)
+
+    # The gradient
+    expected_result = 2.0 * x
+    assert all_almost_equal(func.gradient(x), expected_result,
+                            places=PLACES)
+
+    # The proximal operator
+    sigma = np.random.rand()
+    # Explicit computation: x / (1 + 2*sigma)
```

Comment:
I'd go as far as to say: Comments only for totally non-obvious things, like why to expect a certain behavior, or the result of a complicated computation.